### PR TITLE
Mark Ansible Tower content for move to AWX collection

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
Deprecate this content because it will move to a collection.

https://github.com/ansible/awx/pull/4701

We have not yet released our initial version to awx.awx on Galaxy.

My test releases are at https://galaxy.ansible.com/alancoding/awx

I believe we are fully complete, so we will be pulling the trigger soon (but I want to send out emails, and also have this PR ready when I do)

##### ISSUE TYPE
- Docs Pull Request
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/plugins/doc_fragments/tower.py

##### ADDITIONAL INFORMATION
The new collection has been tested with:

https://github.com/bcoca/ansible-minmal

And everything works fine. We are also running the existing integration tests for the current modules with the new collection.

docs tested with:

```
# current
ansible-doc -t inventory tower
# new collection
ansible-galaxy collection install alancoding.awx -p alan
ANSIBLE_COLLECTIONS_PATHS=alan ansible-doc -M alan/ansible_collections/alancoding/awx/plugins/modules/ -t module tower_host
```

### CODE REVIEW

**Comment by ryanpetrello** [comment]
shipit

**Comment by thedoubl3j** [comment]
👍

**Comment by AlanCoding** [comment]
sanity tests passing locally, will check back in for CI results.

**Comment by AlanCoding** [comment]
I didn't want to commit such a large diff, so I undid the type additions, but I left

https://github.com/ansible/ansible/compare/devel...AlanCoding:tower_deprecate_cherry_pick?expand=1

hanging around, and I've saved this to be eventually cherry-picked into the actual collection.

**Comment by AlanCoding** [comment]
ok good, the tests passed with that. I'm happy with this in its current state now.

**Review by thedoubl3j — APPROVED** [review]
shipit

File: lib/ansible/plugins/inventory/tower.py
Comment (acozine) [inline lib/ansible/plugins/inventory/tower.py:24]: This is a good test change for the new docs pipeline. Can we hold this PR until #59761 is ready?

**Review by acozine — CHANGES_REQUESTED** [review]
We need to be ready to publish collection-based docs before we merge this.

**Comment by acozine** [comment]
Also, this change marks the modules "deprecated". Should they be marked "migrated" or "graduated" instead?

**Comment by AlanCoding** [comment]
We've released a non-beta version of awx.awx collection now.

https://galaxy.ansible.com/awx/awx

The goal of this 7.0.0 version is to be functionally the same as the modules here. New releases will be coming with new patches soon.

**Comment by ryanpetrello** [comment]
@AlanCoding is this in need of some sort of review from me and others?

**Comment by AlanCoding** [comment]
I don't think so?

But there _is_ a bit of an issue up-in-the-air here. We are using the metadata marker "deprecated", but there is a new one for "migrated_to" (I think, might not be congruent) https://github.com/ansible/ansible/pull/63952

That is reflected in some of the conversation here. But that is extremely new, and we have no precedent for using it in a context like this. I don't know if it's _okay_ to use it in a context like this. But I also don't know if the syntax in this PR currently will be preferred. So I'm waiting for direction on that, if there is anything to do.

I'm happy to try doing the newest and best thing, if we can get definition about what that is.

**Comment by MarkusTeufelberger** [comment]
The `foreman` module is also removed upstream and refers to a collection nowadays, maybe it can serve as an example?

**Comment by AlanCoding** [comment]
that is new to me

https://github.com/ansible/ansible/blob/f9ab9b4d6851488b22ae2a017c13503aef26c7ea/lib/ansible/modules/remote_management/foreman/_foreman.py

looks like it's doing about the same thing as here. But that also happened before any migrated_to option was around.

File: lib/ansible/plugins/inventory/tower.py
Comment (AlanCoding) [reply lib/ansible/plugins/inventory/tower.py:24]: That PR looks to be correspondingly blocked on https://github.com/ansible/galaxy/issues/1992

I see that in the foreman case they have their own docs hosting solution, and that's something that we have no intention to bother with for the AWX collection. So it makes sense to me push for the solution there.

**Review by thedoubl3j — APPROVED** [review]
with the addition to the botmeta, this looks good. we are moving to do this for modules that have moved out for partners now as well. lgtm shipit

**Comment by AlanCoding** [comment]
🎉 https://github.com/ansible/ansible/pull/67233 🎉

**Pull request closed without merging**
